sympy/relation/binrel.py

```diff
@@ -0,0 +1,329 @@
+        reflexive = self.is_reflexive
+        if reflexive is None:
+            pass
+        elif reflexive and (lhs == rhs):
+            return True
+        elif not reflexive and (lhs == rhs):
+            return False
+        return None
+
+
+class AppliedBinaryRelation(AppliedPredicate):
+    """
+    The class of expressions resulting from applying ``BinaryRelation``
+    to the arguments.
+
+    This class wraps its argument and remain unevaluated. To evaluate it
+    to boolean value, use :obj:`~.ask()` or :obj:`~.refine()` function.
+
+    Examples
+    ========
+
+    >>> from sympy import cos, sin, Q
+    >>> from sympy.abc import x, y, z
+    >>> eqn1 = Q.eq(sin(x)**2 + cos(x)**2, 1)
+    >>> eqn1
+    sin(x)**2 + cos(x)**2 = 1
+
+    ``.simplify()`` simplifies the relation, but does not evaluate it
+    even if the relation is identical. Also, it does not take assumption
+    into account.
+
+    >>> eqn1.simplify()
+    1 = 1
+    >>> eqn2 = Q.eq(x*y, x*z)
+    >>> eqn2.simplify()
+    x*y = x*z
+
+    ``ask()`` evaluates the relation to boolean value, with taking
+    assumptions into account. If the truth value cannot be determined,
+    it returns ``None``.
+
+    >>> from sympy import ask, Abs
+    >>> eqn3 = Q.eq(Abs(x), x)
+    >>> print(ask(eqn3))
+    None
+    >>> ask(eqn3, Q.positive(x))
+    True
+    >>> ask(eqn3, Q.negative(x))
+    False
+
+    ``refine()`` evaluates the relation to boolean value, with taking
+    assumptions into account. If the truth value cannot be determined,
+    it returns simplified relation.
+
+    >>> from sympy import refine
+    >>> refine(eqn3)
+    Abs(x) = x
+    >>> refine(eqn3, Q.positive(x))
+    True
+    >>> refine(eqn3, Q.negative(x))
+    False
+
+    """
+
+    # will be deleted after _op_priority is removed from SymPy
+    _op_priority = 1000
+
+    @property
+    def lhs(self):
+        """The left-hand side of the relation."""
+        return self.arguments[0]
+
+    @property
+    def rhs(self):
+        """The right-hand side of the relation."""
+        return self.arguments[1]
+
+    @property
+    def reversed(self):
+        """
+        Return the relationship with sides reversed.
+        """
+        revfunc = self.function.reversed
+        if revfunc is None:
+            if self.function.is_symmetric:
+                return self.function(self.lhs, self.rhs)
+            return self
+        return self.function.reversed(self.rhs, self.lhs)
+
+    @property
+    def reversedsign(self):
+        """
+        Return the relationship with signs reversed.
+        """
+        revfunc = self.function.reversed
+        if revfunc is None:
+            return self
+        a, b = self.arguments
+        if not (isinstance(a, BooleanAtom) or isinstance(b, BooleanAtom)):
+            return self.function.reversed(-self.lhs, -self.rhs)
+        else:
+            return self
+
+    @property
+    def canonical(self):
```

Comment:
Shouldn't this method only work if is_symmetric is True?

Reply:
Is the idea that `reversed` can return a different relation, e.g., `LessThan.reversed == GreaterThan`? But the default for `reversed` is None, which doesn't seem to be checked here.